vcs/displayplot.py

```diff
@@ -24,29 +24,37 @@
 #
 #
 from . import VCS_validation_functions
+import sys
 import vcs
 import tempfile
 from .xmldocs import listdoc  # noqa
 from functools import partial
 
+# Import IPython.display and ipywidgets if not imported already
 try:
     import IPython.display
-    HAVE_IPY = True
+except ImportError:
+    pass
+try:
+    import ipywidgets
     try:
         import sidecar
-        HAVE_SIDECAR = True
-    except Exception:
-        HAVE_SIDECAR = False
-    try:
-        import ipywidgets
-        HAVE_IPYWIDGETS = True
-    except Exception:  # no widgets
-        HAVE_IPYWIDGETS = False
-except Exception:  # no IPython
-    HAVE_IPY = False
-    HAVE_SIDECAR = False
-    HAVE_IPYWIDGETS = False
+    except ImportError:
+        pass
+except ImportError:
+    pass
+
+
+def check_module_imported(module):
```

Comment:
You can get rid of all those annoying try/catch and import statements and just use the __import__ method to do it for you (its what gets used under the hood when you use an import statement).

```
def check_module_imported(module):
    if module in sys.modules:
        __import__(module)
        return True
    return False
```

Reply:
The way the script is working now, is that it wants/tries to import all of those modules, but if the module hasn't been installed there's going to be a 'module not found' error. So in the if statement you made, you are importing it, but only if it has already been imported?